core/tests/ts-integration/tests/fees.test.ts

```diff
@@ -143,6 +179,113 @@ testFees('Test fees', () => {
         console.log(`Full report: \n\n${reports.join('\n\n')}`);
     });
 
+    test('Test gas price expected value', async () => {
+        const receiver = ethers.Wallet.createRandom().address;
+        const l1GasPrice = 2_000_000_000n; /// set to 2 gwei
+
+        await setInternalL1GasPrice(alice._providerL2(), {
+            newL1GasPrice: l1GasPrice.toString(),
+            newPubdataPrice: l1GasPrice.toString(),
+            customBaseToken: !isETHBasedChain
+        });
+
+        const receipt = await (
+            await alice.sendTransaction({
+                to: receiver,
+                value: BigInt(1)
+            })
+        ).wait();
+
+        const expectedETHGasPrice =
+            testMaster.environment().l1BatchCommitDataGeneratorMode === DataAvailabityMode.Rollup
+                ? 100_000_000 // 0.1 gwei (the minimum)
+                : 110_000_000; // 0.11 gwei, in validium we need to add compute overhead
+        const expectedCustomGasPrice = expectedETHGasPrice * 0.314;
+
+        if (isETHBasedChain) {
+            expect(receipt.gasPrice).toBe(BigInt(expectedETHGasPrice));
+        } else {
+            expect(receipt.gasPrice).toBe(BigInt(expectedCustomGasPrice));
+        }
+
+        if (!isETHBasedChain) {
+            await setInternalL1GasPrice(alice._providerL2(), {
+                newL1GasPrice: l1GasPrice.toString(),
+                newPubdataPrice: l1GasPrice.toString(),
+                customBaseToken: true,
+                externalPriceApiClientForcedNumerator: 271,
+                externalPriceApiClientForcedDenominator: 100
+            });
+
+            const receipt2 = await (
+                await alice.sendTransaction({
+                    to: receiver,
+                    value: BigInt(1)
+                })
+            ).wait();
+
+            const expectedCustomGasPrice = expectedETHGasPrice * 2.71;
+            expect(receipt2.gasPrice).toBe(BigInt(expectedCustomGasPrice));
+        }
```

Comment:
Probably I miss something, but what's the point of this check, i.e. what does it check that is not covered by other checks?

Reply:
We have added this check to confirm the gas price changes when the ratio is changed.

Reply:
Well, the previous assert checks that gas price depends on the ratio. I wouldn't be against this check unless it required server restart and thus adds 30s of testing time. IMO it isn't worth it. @popzxc WDYT?

Reply:
Agreed.

Reply:
done